libs/agno/agno/workflow/v2/loop.py

```diff
@@ -30,43 +39,382 @@ class Loop:
     name: Optional[str] = None
     description: Optional[str] = None
 
-    max_iterations: Optional[int] = None
-    end_condition: Optional[Union[Callable, Awaitable]] = None
+    max_iterations: int = 1  # Default to 1 iteration as requested
+    end_condition: Optional[Callable[[List[StepOutput]], bool]] = None
```

Comment:
What is the use of `bool` as a type here?

Reply:
its the return type of the callable function